plugins/events-backend/api-report.md

```diff
@@ -42,4 +43,15 @@ export class HttpPostIngressEventPublisher implements EventPublisher {
   // (undocumented)
   setEventBroker(eventBroker: EventBroker): Promise<void>;
 }
+
+// @public
+export class InMemoryEventBroker implements EventBroker {
+  constructor(logger: Logger);
+  // (undocumented)
+  publish(params: EventParams): Promise<void>;
+  // (undocumented)
+  subscribe(
+    ...subscribers: Array<EventSubscriber | Array<EventSubscriber>>
+  ): void;
```

Comment:
I've got some slight hesitation whether this is a flexible enough interface to export as what is essentially the `EventsService`. The `publish` method is alright, but for the `subscribe` I'm feeling the `EventSubscriber` interface is perhaps a bit unnecessarily awkward to implement.

Currently you have to implement the `supportsEventTopics` method, but honestly something like this would be enough:

```ts
eventBroker.subscribe({
  topics: ['foo', 'bar'],
  onEvent: ...
})
```

There's ofc also the option to make that a callback if needed to retain the flexibility of the currently implementation. It's a tiny change code wise but it's a bit of a shift in the mental model for how subscribers are created I feel. It's a move away from for example entity providers implementing the `EventSubscriber` interface and instead passing the `EventService` to them and the rest of the wiring happens internally out of sight of the public API.

The other slight nit here is that I think it's a bit unfortunate to spread the subscribers, it makes it a tiny bit easier to register multiple subscribers, but it locks down the method so that we're never able to add any additional parameters. Given that we're already working with an options object it's a very minor downside, but you never know 😅 

All in all I think this can all be done as future deprecations though, which is to say I think we should move forward with this PR without what I'm suggestion here and treat that all as future potential tweaks.

Reply:
The interface is already exported. However I feel like I generally agree.

Reply:
I think along the same lines. We might want to go a few steps further in the future, however I think we can move forward with this change as first step into this direction.